rewrite-java/src/main/java/org/openrewrite/java/internal/template/Substitutions.java

```diff
@@ -126,7 +126,7 @@ private String substituteTypedPattern(String key, int index, TemplateParameterPa
             if (arrayType.getElemType() instanceof JavaType.Primitive) {
                 s += ((JavaType.Primitive) arrayType.getElemType()).getKeyword();
             } else if (arrayType.getElemType() instanceof JavaType.FullyQualified) {
-                s += ((JavaType.FullyQualified) arrayType.getElemType()).getFullyQualifiedName().replace("$", ".");
+                s += TypeUtils.toFullyQualifiedName(((JavaType.FullyQualified) arrayType.getElemType()).getFullyQualifiedName());
```

Comment:
A `JavaType.FullyQualified` instance currently typically uses the [binary name](https://docs.oracle.com/javase/specs/jls/se8/html/jls-13.html#jls-13.1), which in the compiler's AST is represented by `Symbol#flatName()`.

I think that instead of changing and using `TypeUtils.toFullyQualifiedName()` to get the fully qualified name of a instances implementing `JavaType.FullyQualified` we would be better off either changing the internal representation or deriving the name from the current representation.

Changing the current internal representation is trickier (with regards to backward compatibility), but adding a utility that can return the "source name" for a `JavaType.FullyQualified` instance should be possible by combining what is returned by `getOwningClass()` and by `getFullyQualifiedName()`.

I am saying this because a class could also be called `Foo$Bar` if you wanted to and that is another `$` we shouldn't be replacing.